### PR TITLE
Remove pending requests after the sequence number commit (instead of …

### PR DESCRIPTION
**Pull request by yontyon**
…execution)

The execution is a detemenistic part in the protocol that is not depended on other replicas data. Thus it make sense not to take the execution time in account for complaining the primary.
In this PR we remove the pending request after the commit instead of after the execution.
We expect this change to stablize VC and prevent unnecessary VCs

### CODE REVIEW

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 1]
```diff
@@ -4011,7 +4011,24 @@ void ReplicaImp::executeRequestsAndSendResponses(PrePrepareMsg *ppMsg,
...
     }
-    clientsManager->removePendingRequestOfClient(req.clientId);
+  }
+}
+
+void ReplicaImp::tryToRemovePendingRequestsForSeqNum(SeqNum seqNum) {
+  SCOPED_MDC_SEQ_NUM(std::to_string(seqNum));
+  SeqNumInfo &seqNumInfo = mainLog->get(seqNum);
+  PrePrepareMsg *prePrepareMsg = seqNumInfo.getPrePrepareMsg();
+  if (prePrepareMsg == nullptr) return;
```
Comment (HristoStaykov) [inline bftengine/src/bftengine/ReplicaImp.cpp:4021]: Though we have prior checks and we shouldn't reach this situation, might be it is better to log a warning message as definite indication of a bug.

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 2]
```diff
@@ -4011,7 +4011,24 @@ void ReplicaImp::executeRequestsAndSendResponses(PrePrepareMsg *ppMsg,
...
+  LOG_INFO(GL, "clear pending requests" << KVLOG(seqNum));
+
+  RequestsIterator reqIter(prePrepareMsg);
+  char *requestBody = nullptr;
+  while (reqIter.getAndGoToNext(requestBody)) {
+    ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
+    if (!clientsManager->isValidClient(req.clientProxyId())) continue;
```
Comment (HristoStaykov) [inline bftengine/src/bftengine/ReplicaImp.cpp:4028]: Same as above, might be we should warn that something wrong has happened here.

**Review by HristoStaykov — APPROVED** [review]
LGTM! great work @yontyon ! Just some minor recommendations.

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 1 above)
Comment (yontyon) [reply bftengine/src/bftengine/ReplicaImp.cpp:4021]: I will add this in a future PR

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 2 above)
Comment (yontyon) [reply bftengine/src/bftengine/ReplicaImp.cpp:4028]: This a valid situation, for example, with NOOP commnads

**Pull request merged**